app/controllers/users_controller.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+class UsersController < ApplicationController
+  include UsersHelper
+  include ApplicationHelper
+
+  def show
+    @user = User.find(params[:id])
+    @previous = @user.past_events
+    @upcoming = @user.upcoming_events
+    @previous_created = @user.past_created_events
+    @upcoming_created = @user.upcoming_created_events
```

Comment:
**`OPTIONAL`**
You can rewrite those lines like this
```rb
@user.events.previous_events
@user.attended_events.previous_events
....
```

Reply:
Yes, you are right. it would be better if we use methods directly. :+1: :pray:

Reply:
Thanks for your suggestion. :pray: 
Changes can be seen here: e79ae1857e3809e3c3c3cbe1ceedec2a74b36a63 :eyes: